### DISCUSSION TITLE
Federated Inheritance

### DISCUSSION THREAD

**Discussion by m1sterc001guy** [fedimint/fedimint#1665] [Fedimint Modules]
Traditionally, inheritance has relied on single points of failure (lawyers/executors) to distribute wealth when someone passes away. Executors that are comfortable with breaking the law have the ability to steal and manipulate fortunes that are meant for other people.

Inheritance planning in Bitcoin can be tricky. Self-custody solutions need to balance the ability for someone to access their coins while they're alive, but allow their descendants to retrieve their coins once they're dead. One popular solution is "leaving a trail of bread crumbs" where the deceased leaves behind instructions/clues that will allow their descendants to reconstruct how to retrieve the coins.

Time locks and covenants (like CTV) offer new protection mechanisms that could be useful in inheritance planning. Time locks allow for a UTXO to be locked until a certain block height or wall clock time at which point the UTXO can be spent. On-chain enforced covenants allow more robust expression for how a UTXO can be spent, but currently require a soft-fork to be activated in Bitcoin. Users can also simply pre-sign transactions and delete the key for the spent coins (to simulate covenant-like behavior), but this introduces a new problem on how to back-up the pre-signed transactions.

I think Fedimint could offer a new trade-off balance for inheritance planning. Users could "lock" their coins in an inheritance contract that is enforced by a quorum of guardians. The inheritance contract could specify Bitcoin addresses, dispersal amounts, and/or a time lock at which point the coins could be "un-locked" and reclaimed onchain. Additionally, the Fedimint guardians could potentially use distinct oracles for determining if someone is dead (potentially by querying some 3rd party API?), at which point the contract could be automatically executed and the funds dispersed to the descendants (this might not be a good idea if the Oracles are not trustworthy).

The main advantage of using a Federation for enforcing an inheritance contract is eliminating the single point of trust with the traditional lawyer/executor, but having a more automated plan than pure self-custodial Bitcoin. The obvious down-side of using a Federation for inheritance planning is the need to entrust a large some of money with a federation.

Thoughts?

**Comment by jkitman** [comment]
Storing significant savings long-term entirely in Fedimint probably doesn't make sense...at least not for now, given the risk trade-offs.

I think companies like Unchained or Casa have elegant solutions for this already.  Fedimint could have a module that is similar, such as threshold-encrypting one of the multisig keys.  You could even back up PSBTs to the Fedimint in case of key loss / simulating a covenant.

**Reply by ecurrencyhodler** [reply]
Yeah solving for bitcoin inheritance comes down to 2 aspects:
1. Legal considerations + estate planning
2. Making sure intended inheritance recipients are able to receive/access the bitcoin

2 is already possible and can be done in a trust-minimized way using multisig already. But 1 is what really makes bitcoin inheritance difficult. m1sterc001guy's scenario of creating an inheritance contract that pays out x btc over x period of time is essentially what an estate planner or executor would do. 

In reality, the fedimint is in full possession and custody of the bitcoin that someone else has legal rights to which ends up being a tricky legal situation. I can see offering an inheritance solution as a paid service that fully regulated fedimints offer their users though.

One last thought is the dedicated bitcoin address. This in theory sounds nice but is risky to implement. There is no guarantee that the intended recipient will have access to the keys to the bitcoin address in the future. It is much better to hire or appoint an estate planner instead who can flexibly distribute bitcoin as needed or scheduled.
